tensorflow/c/eager/gradient_checker.cc

```diff
@@ -191,6 +191,8 @@ Status CalcNumericalGrad(AbstractContext* ctx, Model forward,
     dtheta_approx[i] = grad_data[0];
   }
 
+  // Restore the inputs
+  inputs[input_index] = theta;
```

Comment:
> It may be cleaner to create a shallow copy of inputs that we can update instead of updating in-place and restoring.

I've already though about it. However, I could not find a way to create a shallow copy. We have to create a `std::vector<AbstractTensorHandle*>` because another span still modifies the input. In addtion, with `absl::Span<AbstractTensorHandle* const> inputs`, I could not create a `std::vector<AbstractTensorHandle*>` because there is a loss of constness ( `abseil` seems to be very strict with this behavior )

Reply:
Hmm, that's interesting. So something like

```
Status CalcNumericalGrad(AbstractContext* ctx, Model forward,
                         absl::Span<AbstractTensorHandle* const> inputs,
                         int input_index, bool use_function,
                         AbstractTensorHandle** numerical_grad) {
  std::vector<AbstractTensorHandle*> inputs_copy(inputs.size());
  for (int i = 0; i < inputs.size(); i++) {
    inputs_copy[i] = inputs[i];
  }
```

doesn't work?

Reply:
It works but it seems sub-optimal 😕